### PR TITLE
Tolerations modal

### PR DESCRIPTION
**Pull request by avivtur**
## 📝 Description

Adding tolerations modal to utils/components and consume it from VM details page under the scheduling section

## 🎥 Demo

https://user-images.githubusercontent.com/67270715/163968099-0045c276-0d7d-4c7c-8897-043d81b1cfcc.mp4

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

File: src/utils/components/TolerationsModal/hooks/useNodeTaintQualifier.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+              if (taint?.value && value) {
+                return taint.key === key && taint.value === value && taint.effect === effect;
+              }
+              return taint.key === key && taint.effect === effect;
+            }),
+          )
+        ) {
+          suitableNodes.push(node);
+        }
+      });
```
Comment (metalice) [inline src/utils/components/TolerationsModal/hooks/useNodeTaintQualifier.ts:35]: can be changed to map or filter?

Comment (metalice) [reply src/utils/components/TolerationsModal/hooks/useNodeTaintQualifier.ts:35]: perhaps even useffect isnt needed.

Comment (avivtur) [reply src/utils/components/TolerationsModal/hooks/useNodeTaintQualifier.ts]: changed thanks for the tip :)

File: src/utils/components/TolerationsModal/TolerationsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  const qualifiedNodes = getNodeTaintQualifier(nodes, nodesLoaded, tolerationsLabels);
+
+  const onSelectorLabelAdd = () =>
+    onTolerationAdd({ id: null, key: '', value: '', effect: TOLERATIONS_EFFECTS[0] });
+
+  const updatedVirtualMachine = React.useMemo(() => {
+    const updatedVM = produce<V1VirtualMachine>(vm, (vmDraft: V1VirtualMachine) => {
+      ensurePath(vmDraft, ['spec.template.template.spec.tolerations']);
+      if (!vmDraft.spec.template.spec.tolerations) {
```
Comment (metalice) [inline src/utils/components/TolerationsModal/TolerationsModal.tsx:60]: add .? will still return undefined but without a crash

Comment (metalice) [reply src/utils/components/TolerationsModal/TolerationsModal.tsx:60]: actuall is it needed? in line 71 can be vmDraft.spec.template.spec.tolerations = updatedTolerations || []

Comment (avivtur) [reply src/utils/components/TolerationsModal/TolerationsModal.tsx]: added || [] to line 61 where I map the labels, this should have the same affect as suggested :)

**Pull request merged**